### DISCUSSION TITLE
How can we use nuxt3 to share data between the server and the client and make the data persistent?

### DISCUSSION THREAD

**Discussion by tanglongyan** [nuxt/nuxt#26834] [Questions]
### Environment

none

### Reproduction

How can we use nuxt3 to share data between the server and the client and make the data persistent? 

### Describe the bug

The shared state in the useState api is lost after the page is refreshed. Even if I use the third-party state persistence tool pinia, piniaPersistedstate plug-in, I found that its state cannot get the value on the server side.（Is it possible to share a data between the server and the client and persist it only through the useCookie api?）

### Additional context

none

**Comment by huang-julien** [comment]
Hey :wave:

UseState shouldn't be flushed after a page change. You probably have a piece of code that reset it on page load or change

**Reply by tanglongyan** [reply]
I am saying that the data saved in useState will lose the changed values in the shared data when the page is refreshed in the browser

**Reply by tanglongyan** [reply]
How can we use nuxt3 to share data between the server and the client and make the data persistent?

**Reply by huang-julien** [reply]
useState is no meant to be shared across requests. You will need to find another solution such as having a global variable/cache in your server.

You could probably use cookies to keep a persistent state but there is some (size) limitations.

**Comment by liaow-ww** [comment]
你的问题是在服务端无法获取到根据组件内逻辑修改后更新后的值？如果是存在pinia里的值，需要在服务端中获取，建议使用cookie，因为服务端请求无法获取storage的值，客户端时可以
